File: Week_04/45/jump_game_ii.go
package main

func jump(nums []int) int {
	steps, end, maxPos := 0, 0, 0
	for i := 0; i < len(nums)-1; i++ {
		maxPos = max(maxPos, i+nums[i])
		if i == end {
			end = maxPos
			steps++
		}
	}
	return steps
}

func max(x, y int) int {
	if x > y {
		return x
	}
	return y
}

File: Week_04/45/jump_game_ii.py
from typing import List


class Solution:
    def jump(self, nums: List[int]) -> int:
        steps = end = max_pos = 0
        for i in range(0, len(nums) - 1):
            max_pos = max(max_pos, i + nums[i])
            if i == end:
                steps += 1
                end = max_pos
        return steps
